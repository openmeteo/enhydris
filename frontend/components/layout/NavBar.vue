<template lang="html">
  <div>
    <nav class="navbar is-dark" role="navigation" aria-label="main navigation">
      <div class="navbar-brand">
        <a
          :class="{ 'is-active': isOpen }"
          role="button"
          class="navbar-burger burger"
          aria-label="menu"
          aria-expanded="false"
          data-target="navbarBasicExample"
          @click="isOpen = !isOpen"
        >
          <span aria-hidden="true"></span>
          <span aria-hidden="true"></span>
          <span aria-hidden="true"></span>
        </a>
      </div>

      <div class="navbar-menu" :class="{ 'is-active': isOpen }">
        <div class="navbar-start">
          <a
            id="home"
            class="navbar-item"
            :href="currentLanguage != 'en' ? currentLanguage : ''"
          >
            {{ $t("home") }}
          </a>
        </div>
        <div class="navbar-end">
          <LangPicker />
        </div>
      </div>
    </nav>
  </div>
</template>

<script>
import LangPicker from "@/components/layout/LangPicker";

export default {
  name: "NavBar",
  components: {
    LangPicker
  },
  data() {
    return {
      isOpen: false
    };
  },
  computed: {
    currentLanguage() {
      return this.$i18n.locale || "en";
    }
  }
};
</script>

<style scope>
.navbar {
  padding-top: 5px;
  padding-bottom: 5px;
}
</style>
