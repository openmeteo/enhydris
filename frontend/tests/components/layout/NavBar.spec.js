import { shallowMount } from "@vue/test-utils";
import NavBar from "@/components/layout/NavBar";

describe("NavBar with locale en", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(NavBar, {
      mocks: {
        $t: msg => msg,
        $i18n: { locale: "en" }
      }
    });
  });

  it("has default language en", () => {
    expect(wrapper.vm.currentLanguage).toEqual("en");
  });

  it("renders $t('home')", () => {
    expect(wrapper.find("#home").element.text).toContain("home");
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
