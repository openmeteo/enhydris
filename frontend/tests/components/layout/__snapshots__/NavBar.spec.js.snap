// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar with locale en renders correctly 1`] = `
<div>
  <nav
    aria-label="main navigation"
    class="navbar is-dark"
    role="navigation"
  >
    <div
      class="navbar-brand"
    >
      <a
        aria-expanded="false"
        aria-label="menu"
        class="navbar-burger burger"
        data-target="navbarBasicExample"
        role="button"
      >
        <span
          aria-hidden="true"
        />
         
        <span
          aria-hidden="true"
        />
         
        <span
          aria-hidden="true"
        />
      </a>
    </div>
     
    <div
      class="navbar-menu"
    >
      <div
        class="navbar-start"
      >
        <a
          class="navbar-item"
          href=""
          id="home"
        >
          
          home
        
        </a>
      </div>
       
      <div
        class="navbar-end"
      >
        <langpicker-stub />
      </div>
    </div>
  </nav>
</div>
`;
